![](img/hecatombLogo.png)

A hecatomb is a great sacrifice or an extensive loss. 
Heactomb the software empowers an analyst to make data driven decisions to 'sacrifice' false-positive viral reads from 
metagenomes to enrich for true-positive viral reads. 
This process frequently results in a great loss of suspected viral sequences / contigs.

![](img/hecatombBanner.png)

Hecatomb was developed in response to the challenges associated with the detection of viral sequences in metagenomes. 
Virus detection or virome profiling is typically performed on samples containing extensive host nucleic acid (e.g. a 
tissue biopsy) or nucleic acid from a variety of other organisms such as bacteria, fungi and archaea from soil samples 
or bacteria, fungi, archaea and food from mammalian stool samples. 
All of these non-viral nucleic acid types constitute the background and present a variety of issues that must be 
evaluated in order to confidently detect viral sequences.

Detection and quantitative analysis of viral sequences from mixed metagenomic data has other unique challenges. 
Viruses have a propensity to rapidly acquire mutations and evolve to be highly diverged from reference database sequences.
Coding and non-coding regions of viral genomes have disparate challenges (e.g. long-terminal 5' and 3' repeats).
Viruses are further clssified as either single or double stranded, and either DNA or RNA.
False-positives are rampant in many studies due to significant sequence homology between viruses and other domains of life.

Hecatomb utilises a unique search strategy to address these issues and enable researchers to accurately identify and quantify 
viruses in their metagenome samples.
